lib/pure/stats.nim

```diff
@@ -53,6 +53,37 @@ runnableExamples:
   doAssert statistics.kurtosis() ~= -1.0
   doAssert statistics.kurtosisS() ~= -0.7000000000000008
 
+  var 
+    ASv = [7,4,6,3,9,1]        #  sorted   1 3 4 6 7 9
+    A   : array[len(ASv),int]
+
+  func myCmp(x,y:int):int =
+    if x==y : 0 elif x<y : -1 else: 1
+
+  A= ASv;  doAssert quickSelect(A, 2, 5, 2, myCmp) == 1  # smallest element
+  A= ASv;  doAssert quickSelect(A, 2, 5, 5, myCmp) == 9  # largest element
+  #
+  A= ASv;  doAssert median(A,0,5,myCmp)      == 5.0
+  A= ASv;  doAssert median_low(A,0,5,myCmp)  == 4
+  A= ASv;  doAssert median_high(A,0,5,myCmp) == 6
```

Comment:
expose only a single API `median` with a param medianMode, an enum

Reply:
That's difficult. 
```nim
proc median*[T: SomeNumber]
```
return a **float** value for all ``T`` while   ``median_low[T]/median_high[T]`` return a value of type ``T``.
``median_low/median_high`` could be joined into a single proc - but  would should it be called?